backend/fileSystem.js

```diff
@@ -145,6 +147,84 @@ class FileSystemBackend {
         const match = await this.getPackageByName(packageName);
         return match ? match.versions[version] : null;
     }
+    async searchPackages(properties) {
+        // Retrieve listing of all packages
+        await this.syncMetadata();
+        const packages = Object.keys(this._metadata).map(key => this._metadata[key]);
+
+        // Begin with all packages
+        let result = packages;
+
+        // Filter by search query
+        if (typeof properties.query === 'string' && properties.query.length > 0) {
+            // If only searching for names, it's possible to more efficiently search the whole package array
+            let searchProperties = packages;
+
+            // If doing an advanced search with additional properties, make a new array of package properties
+            if (properties.property.length !== 1 || properties.property[0] !== 'name') {
+                // Extract searchable properties
+                searchProperties = packages.map((pack) => {
+                    // Get latest
+                    const latest = pack.versions[pack['dist-tags'].latest];
+
+                    // Return selected properties
+                    return {
+                        name: pack.name,
+                        description: latest.description || pack.description,
+                        author: latest._npmUser.name,
+                        readme: latest.readme || pack.readme,
+                        keywords: latest.keywords,
+                    };
+                });
+            }
+
+            // Set options for searching
+            const options = {
```

Comment:
`shouldSort` should always be `true` and `keys` should always be its picked values. That leaves `maxPatternLength` which specifies how many characters to stop searching for (to avoid bad inefficiencies), and `threshold` which defines the score at which it no longer considers it a match. I could move these two into the configuration file, but they seem like very reasonable defaults and I cannot think of a situation where they would ever need to be changed. For the threshold value, we already have a cutoff so that only the requested number of results are returned, and anything less matched than 0.6 is likely totally unrelated to the requested results. For the max pattern length, I doubt someone would ever need to search past 32 characters and require that those characters past the first 32 count towards the matching characteristics, and below 32 doesn't affect performance much. But I can add these if you still want, but I think it would just increase code complexity without any benefit because they are very reasonable defaults.